in_toto/model.go

```diff
@@ -704,6 +677,105 @@ func validateLayout(layout Layout) error {
 	return nil
 }
 
+type Metadata interface {
+	Sign(Key) error
+	VerifySignature(Key) error
+	GetPayload() any
```

Comment:
not sure you can change the interface to return an error also.

Reply:
I think I'd prefer to rework how the required field validations happens to avoid using sync.Once and retain GetPayload without return an error.